pkg/resource/deploy/step_generator.go

```diff
@@ -555,7 +555,79 @@ func (sg *stepGenerator) GenerateDeletes() []Step {
 			}
 		}
 	}
-	return dels
+
+	if len(targets) > 0 {
+		logging.V(7).Infof("Planner was asked to only delete '%v'", targets)
+
+		resourcesToDelete := make(map[resource.URN]bool)
+
+		for _, target := range targets {
+			current, has := sg.plan.olds[target]
+			if !has {
+				logging.V(7).Infof("Resource to delete (%v) could not be found in the stack.", target)
+				if strings.Contains(string(target), "$") {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundError(), target)
+				} else {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundDidYouForgetError(), target)
+				}
+				return nil, result.Bail()
+			}
+
+			resourcesToDelete[target] = true
+
+			// the item the user is asking to destroy may cause downstream replacements.  Clean those up
+			// as well. Use the standard delete-before-replace computation to determine the minimal
+			// set of downstream resources that are affected.
+			deps, res := sg.calculateDependentReplacements(current)
```

Comment:
I am a _bit_ nervous about deleting resources that the user did not explicitly request. We do need to delete these resources, but I wonder if we should make the user explicitly list them as well as the root resource. @lukehoban thoughts?

Reply:
I'm personally 100% ok with it.  By default we let users know what will happen and there will be the set of items we think are necessary to delete here.  I think the confirmation already comes from having to approve this.  I'm not sure i think it's pleasant to have to force them to explicitly enumerate these all on the command line. 

But i'm open to alternate views here.  @lukehoban any concerns on your part?

Reply:
I'm fine with the preview being the way you can approve it.  We should make sure the help text for this command is clear that if you specify a target you are specifying that you want to delete "only that resources and any other resources that must be deleted as a result".